packages/react-components/react-tree/src/components/BaseTreeItem/useBaseTreeItem.ts

```diff
@@ -32,33 +26,41 @@ export const useBaseTreeItem_unstable = (
   const isBranch = typeof ariaOwns === 'string';
   const open = useTreeContext_unstable(ctx => isBranch && ctx.openSubtrees.includes(ariaOwns!));
 
-  const handleClick = useEventCallback((event: React.MouseEvent<BaseTreeItemElementIntersection>) => {
+  const handleClick = useEventCallback((event: React.MouseEvent<HTMLDivElement>) => {
     if (isBranch) {
       requestOpenChange({ event, open: !open, type: 'click', id: ariaOwns! });
     }
   });
-  const handleArrowRight = (event: React.KeyboardEvent<BaseTreeItemElementIntersection>) => {
+  const handleArrowRight = (event: React.KeyboardEvent<HTMLDivElement>) => {
     if (open && isBranch) {
       focusFirstSubtreeItem(event.currentTarget);
     }
     if (isBranch && !open) {
       requestOpenChange({ event, open: true, type: 'arrowRight', id: ariaOwns! });
     }
   };
-  const handleArrowLeft = (event: React.KeyboardEvent<BaseTreeItemElementIntersection>) => {
+  const handleArrowLeft = (event: React.KeyboardEvent<HTMLDivElement>) => {
     if (!isBranch || !open) {
       focusSubtreeOwnerItem(event.currentTarget);
     }
     if (isBranch && open) {
       requestOpenChange({ event, open: false, type: 'arrowLeft', id: ariaOwns! });
     }
   };
-  const handleKeyDown = useEventCallback((event: React.KeyboardEvent<BaseTreeItemElementIntersection>) => {
+  const handleEnter = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (isBranch) {
+      requestOpenChange({ event, open: !open, type: 'enter', id: ariaOwns! });
+    }
+  };
+  const handleKeyDown = useEventCallback((event: React.KeyboardEvent<HTMLDivElement>) => {
     onKeyDown?.(event);
     if (event.isDefaultPrevented()) {
       return;
     }
-    switch (event.code) {
+    switch (event.key) {
+      case Enter: {
+        return handleEnter(event);
```

Comment:
should enter be used to open subtrees? it's not a behaviour I see in the APG example https://www.w3.org/WAI/ARIA/apg/example-index/treeview/treeview-1/treeview-1a.html

Reply:
According to https://www.w3.org/WAI/ARIA/apg/patterns/treeview/, yeah

Enter: activates a node, i.e., performs its default action. For parent nodes, one possible default action is to open or close the node. In single-select trees where selection does not follow focus (see note below), the default action is typically to select the focused node.